apps/api/src/app/workflows-v2/usecases/patch-workflow/patch-workflow.usecase.ts

```diff
@@ -31,20 +31,17 @@ export class PatchWorkflowUsecase {
     });
   }
 
-  private patchWorkflowFields(persistedWorkflow, command: PatchWorkflowCommand) {
-    const transientWorkflow = { ...persistedWorkflow };
-    if (command.active !== undefined) {
+  private patchWorkflowFields(persistedWorkflow, command: PatchWorkflowCommand): NotificationTemplateEntity {
+    const transientWorkflow: NotificationTemplateEntity = { ...persistedWorkflow };
+    if (command.active !== undefined && command.active !== null) {
       // @ts-ignore
       transientWorkflow.active = command.active;
     }
 
     return transientWorkflow;
   }
 
-  private async persistWorkflow(
-    workflowWithIssues: Partial<NotificationTemplateEntity>,
-    userSessionData: UserSessionData
-  ) {
+  private async persistWorkflow(workflowWithIssues: NotificationTemplateEntity, userSessionData: UserSessionData) {
     await this.notificationTemplateRepository.update(
```

Comment:
We should use the `UpdateWorkflow` use-case now (which already behaves as `PATCH` style use-case) to ensure complete validation occurs before reaching the DB layer.

Reply:
In fact, do we even need a `PatchWorkflow` use-case? Can we just call `UpdateWorkflow` directly from the `PATCH /v2/workflow` controller endpoint?

Reply:
Valid question, we could check it, i think the intention was to add gradual update support.
IMO in the end patch-workflow.usecase.ts could be a duplication of update-workflow use case although maybe a bit leaner.

Reply:
we are trying to create a new order, cleaner, we don't want to expose everything at once, stuff has side effects.